src/LanguageServer/Impl/Telemetry.cs

```diff
@@ -21,6 +21,30 @@
 using StreamJsonRpc;
 
 namespace Microsoft.Python.LanguageServer.Implementation {
+    internal class Telemetry {
+        private const string EventPrefix = "python_language_server/";
+        private static readonly string PLSVersion = GetPLSVersion();
+
+        public static TelemetryEvent CreateEvent(string eventName) {
+            var e = new TelemetryEvent {
+                EventName = EventPrefix + eventName,
+            };
+            e.Properties["plsVersion"] = PLSVersion;
```

Comment:
Shouldn't this be in `TelemetryService.SendTelemetry`? I.e.:
```cs
public Task SendTelemetry(TelemetryEvent telemetryEvent) {
    telemetryEvent.Properties["plsVersion"] = PLSVersion;
    _rpc.NotifyWithParameterObjectAsync("telemetry/event", telemetryEvent);
}
```

Reply:
I'm not sure, since it's also being used by Intellicode to send their own telemetry and I'm not sure I want to add properties to their events.

Reply:
Though, including the language server version wholesale would probably not be a bad idea.

Reply:
Intellicode depends on LS, so having information about LS version is also important.

Reply:
Alright, I guess I'll move this out into the service. Should I just overwrite the dictionary's `plsVersion` value (as your suggested code does) or check before setting? (I can't imagine they'd ever try to set it differently, but...)

I'll probably also have to null check too, since currently a null event is allowed to go through, as useless as that is. Maybe it'd be better to just have the service ignore any null events outright.

Reply:
I think we shouldn't allow overriding of our own event values by plugins, this information is important. As for null ref, I'd prefer to fail. As you said, null events are meaningless, and most likely it is a bug.